server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -957,7 +957,7 @@ private boolean invariant() {
             // all tracked shard copies have a corresponding peer-recovery retention lease
             for (final ShardRouting shardRouting : routingTable.assignedShards()) {
                 final CheckpointState cps = checkpoints.get(shardRouting.allocationId().getId());
-                if (cps.tracked && cps.replicated) {
+                if (cps.tracked && cps.replicated && shardRouting.isAssignedToRemoteStoreNode() == false) {
```

Comment:
does `cps.replicated=true` make sure that is assigned to docrep node only ? It would mean we could remove the changes in this line.